### PR TITLE
Connect frontend and backend to allow users to post a listing

### PR DESCRIPTION
**Pull request by Jortiz23**
The bulk of this process took place in the "PostListingDisplay.jsx" file. Added a function to calculate the listing price based on the inputs provided by the user.

Also added tests for the react components. Both snapshot and controlled component testing were included.

### CODE REVIEW

File: src/PostListingDisplay/components/PostListingDisplay.jsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        this.calculateListingPrice = this.calculateListingPrice.bind(this);
+    }
+
+    handleChange(event) {
+        const value = event.target.value;
+        const name = event.target.name;
+        this.setState({
+            [name]: value
+        })
+        this.calculateListingPrice();
```
Comment (shainachen) [inline src/PostListingDisplay/components/PostListingDisplay.jsx:37]: would it be possible to only calculate listing price if there's a change to the relevant fields (price or room type)?

File: src/PostListingDisplay/components/PostListingDisplay.jsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    calculateListingPrice(){
+        const numSingles = this.state.numSingles;
+        const singlePrice = this.state.singlePrice;
+        const numShared = this.state.numShared;
+        const sharedPrice = this.state.sharedPrice;
+        const listingPriceValue = numSingles * singlePrice + numShared * sharedPrice;
+        if (isNaN(listingPriceValue)) {
+            this.setState({
+                listingPrice:'0'
+            });
```
Comment (shainachen) [inline src/PostListingDisplay/components/PostListingDisplay.jsx:49]: Do we want to set listing price to 0 here, or should we be keeping it empty so that it won't let you post the form unless there is a valid listing price?

Comment (Jortiz23) [reply src/PostListingDisplay/components/PostListingDisplay.jsx:49]: Yeah we want it empty. Fixed

File: src/PostListingDisplay/components/PostListingDisplay.jsx (hunk 1 above)
Comment (Jortiz23) [reply src/PostListingDisplay/components/PostListingDisplay.jsx]: Done

File: .gitignore [hunk 3]
```diff
@@ -1,3 +1,3 @@
 # Ignore generated Maven build files
 target/
-node_modules/
\ No newline at end of file
+node_modules/
```
Comment (NathanBricault) [inline .gitignore:3]: Is this a whitespace change or a newline? Adds noise to the edit history, I would remove

File: package.json [hunk 4]
```diff
@@ -2,19 +2,21 @@
   "name": "roomies-project",
   "version": "0.1.0",
   "private": true,
+  "proxy": "http://localhost:8080",
```
Comment (NathanBricault) [inline package.json:5]: Is this a setting that should be committed? Or is this something used for local testing?

File: src/App.test.js [hunk 5]
```diff
@@ -6,4 +6,4 @@ test('renders learn react link', () => {
   const { getByText } = render(<App />);
   const linkElement = getByText(/learn react/i);
   expect(linkElement).toBeInTheDocument();
-});
+});
```
Comment (NathanBricault) [inline src/App.test.js:9]: Same comment as earlier - this doesn't seem like an intentional change

File: src/PostListingDisplay/components/PostListingDisplay.jsx [hunk 6]
```diff
@@ -0,0 +1,98 @@
+import React, { Component } from 'react';
+
+import InputField from './InputField';
+
+import '../styles/PostListingDisplay.css';
+
+/**Component class that displays the input fields for a user to post a listing */
```
Comment (NathanBricault) [inline src/PostListingDisplay/components/PostListingDisplay.jsx:7]: nit: space after **

File: src/PostListingDisplay/components/PostListingDisplay.jsx [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+        this.handleChange = this.handleChange.bind(this);
+        this.calculateListingPrice = this.calculateListingPrice.bind(this);
+    }
+
+    handleChange(event) {
+        const value = event.target.value;
+        const name = event.target.name;
+        this.setState({ [name]: value }, () => {
+            if(name === "numSingles" || name === "singlePrice" || name === "numShared" || name === "sharedPrice"){
+                this.calculateListingPrice();
```
Comment (NathanBricault) [inline src/PostListingDisplay/components/PostListingDisplay.jsx:36]: I think this should either be split into two steps (pure calculation and memory update), or renamed to update so it's clear that it has side effects.

File: src/PostListingDisplay/components/PostListingDisplay.jsx [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+
+        this.handleChange = this.handleChange.bind(this);
+        this.calculateListingPrice = this.calculateListingPrice.bind(this);
+    }
+
+    handleChange(event) {
+        const value = event.target.value;
+        const name = event.target.name;
+        this.setState({ [name]: value }, () => {
+            if(name === "numSingles" || name === "singlePrice" || name === "numShared" || name === "sharedPrice"){
```
Comment (NathanBricault) [inline src/PostListingDisplay/components/PostListingDisplay.jsx:35]: Rather than manually checking which fields are being updated, is there a reason not to use a dedicated handler for mutations to those fields specifically?

File: .gitignore (hunk 3 above)
Comment (Jortiz23) [reply .gitignore:3]: Done

File: package.json (hunk 4 above)
Comment (Jortiz23) [reply package.json:5]: It's for local testing. I've removed it

File: src/App.test.js (hunk 5 above)
Comment (Jortiz23) [reply src/App.test.js:9]: Done

File: src/PostListingDisplay/components/PostListingDisplay.jsx (hunk 6 above)
Comment (Jortiz23) [reply src/PostListingDisplay/components/PostListingDisplay.jsx:7]: Done

File: src/PostListingDisplay/components/PostListingDisplay.jsx (hunk 8 above)
Comment (Jortiz23) [reply src/PostListingDisplay/components/PostListingDisplay.jsx:35]: I hadn't thought of that, but that is much better. I added a function called handlePriceChange() that calls handlechange and calculateListingPrice and changed the onchange function for the 4 price determining inputs to handlePriceChange()

File: src/PostListingDisplay/components/PostListingDisplay.jsx (hunk 7 above)
Comment (Jortiz23) [reply src/PostListingDisplay/components/PostListingDisplay.jsx:36]: I have split it up into updateListingPrice and calculateListingPrice. updateListingPrice calls calculateListingPrice to get what the listing price should be and then updates the state.

File: src/PostListingDisplay/components/InputField.jsx [hunk 9]
```diff
@@ -27,9 +37,9 @@ function setDefault(type){
                 day = '0' + day;
             } 
             if(month < 10) {
-                month = '0' + month;
+                month = '0' + (month+1);
```
Comment (NathanBricault) [inline src/PostListingDisplay/components/InputField.jsx:40]: This seems wrong - if month values are returned 0-11, you'll print month 9 as 010, and month 10 as 10?

File: src/PostListingDisplay/components/PostListingDisplay.jsx [hunk 10]
```diff
@@ -62,22 +74,22 @@ class PostListingDisplay extends Component {
                     <h1>Post a Listing</h1>
                 </div>
 
-                <form action="/listings" method="POST">
+                <form data-testid="form" action="/listings" method="POST">
```
Comment (NathanBricault) [inline src/PostListingDisplay/components/PostListingDisplay.jsx:77]: Is this intentional? Testid is not an name I would expect in prod code.

File: src/test/react/SnapshotComponentTesting/PostCommentsDisplay.test.js [hunk 11]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import renderer from 'react-test-renderer';
+import PostCommentsDisplay from '../../../PostCommentsDisplay/components/PostCommentsDisplay.jsx';
+import CommentEditor from '../../../PostCommentsDisplay/components/CommentEditor.jsx';
+import Comment from '../../../PostCommentsDisplay/components/Comment.jsx';
+
+test('PostCommentDisplay renders correctly', () => {
```
Comment (NathanBricault) [inline src/test/react/SnapshotComponentTesting/PostCommentsDisplay.test.js:7]: Why are comment tests here? Seems like this should be part of a separate pull request - this one is nominally about listings not comments.

File: src/PostListingDisplay/components/InputField.jsx (hunk 9 above)
Comment (Jortiz23) [reply src/PostListingDisplay/components/InputField.jsx:40]: You're right. I put the "+1" in the wrong spot. Moved it to where month first gets initialized.

File: src/PostListingDisplay/components/PostListingDisplay.jsx (hunk 10 above)
Comment (Jortiz23) [reply src/PostListingDisplay/components/PostListingDisplay.jsx:77]: That should not be there. Thank you for pointing that out

File: src/test/react/SnapshotComponentTesting/PostCommentsDisplay.test.js (hunk 11 above)
Comment (Jortiz23) [reply src/test/react/SnapshotComponentTesting/PostCommentsDisplay.test.js]: Removed comments tests. Will add them back in when appropriate
